windows/src/desktop/kmshell/main/Keyman.System.DownloadUpdate.pas

```diff
@@ -0,0 +1,203 @@
+(*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ *)
+unit Keyman.System.DownloadUpdate;
+
+interface
+uses
+  System.Classes,
+  System.SysUtils,
+  httpuploader,
+  Keyman.System.UpdateCheckResponse,
+  KeymanPaths,
+  OnlineUpdateCheck;
+
+const
+  DaysBetweenCheckingForUpdates: Integer = 7; // Days between checking for updates
+
+type
+  TDownloadUpdateParams = record
+    TotalSize: Integer;
+    TotalDownloads: Integer;
+    StartPosition: Integer;
+  end;
+
+  TDownloadUpdate = class
+  private
+    FShowErrors: Boolean;
+    FDownload: TDownloadUpdateParams;
+    (**
+     *
+     * Performs updates download in the background, without displaying a GUI
+     * progress bar.
+     * @params  SavePath  The path where the downloaded files will be saved.
+     *          Result    A Boolean value indicating the overall result of the
+     *          download process.
+     *)
+    procedure DoDownloadUpdates(SavePath: string; Params: TUpdateCheckResponse;  var Result: Boolean);
+
+  public
+
+    constructor Create;
+    destructor Destroy; override;
+    (**
+     * Performs updates download in the background, without displaying a GUI
+     * progress bar. This function is similar to DownloadUpdates, but it runs in
+     * the background.
```

Comment:
This is the function comment for `DownloadUpdates` so I don't understand this?

Reply:
Yes very confusing. This comment needs updating or removing it is from when this function was the unit `TOnlineUpdateCheck.DownloadUpdates` and it was actually called something like DownloadBackgroundUpdates. I ended up creating a new unit `Keyman.System.DownloadUpdate` for greater clarity.